npm/react/plugins/react-scripts/index.js

```diff
@@ -1,26 +1,22 @@
+const wrapDevServer = require('../utils/wrap-devserver')
 const { startDevServer } = require('@cypress/webpack-dev-server')
 const findReactScriptsWebpackConfig = require('./findReactScriptsWebpackConfig')
 
-module.exports = (
-  on,
-  config, {
-    webpackConfigPath,
-  } = {
-    webpackConfigPath: 'react-scripts/config/webpack.config',
-  },
-) => {
-  on('dev-server:start', async (options) => {
-    return startDevServer({
-      options,
-      webpackConfig: findReactScriptsWebpackConfig(config, {
-        webpackConfigPath,
-      }),
-    })
+function startReactScriptsDevServer (options, {
+  webpackConfigPath,
+} = {
+  webpackConfigPath: 'react-scripts/config/webpack.config',
+}) {
+  return startDevServer({
```

Comment:
The `webpack-dev-server` function should follow one of the pairs of function signatures defined in [notion](https://www.notion.so/cypressdx/cypress-config-js-3a0fb1ca63ac4843b3cbe3b56d3bcd9a#9dd640146b0741a28c7f4dd5eeff8cd4).

Reply:
I believe it does. 
NOTE: this function is not the exported one but only the devServer function itself.

Reply:
The [StartDevServer type](https://github.com/cypress-io/cypress/blob/0dfec805f47c0afbbfbb0dbcbab351e6c13e829d/npm/webpack-dev-server/src/startServer.ts#L7) that the [webpack-dev-server](https://github.com/cypress-io/cypress/blob/develop/npm/webpack-dev-server/src/index.ts#L17) module uses is not the API outlined in the notion document. Can we also make the `webpack-dev-server` signatures consistent with the others?

Reply:
`webpackDevServer` and the plugins like this one are two different levels of abstraction.

One can be used on it's own, with very little options to customize (the plugins like this one).
The other needs us to provide a complete configuration of options.

These "plugins" are devServers pre-configured for a specific common setup (CRA, CRACO, NEXT).
Since they are a level of abstraction above the "real" dev server, webpack or vite, we try to name them differently. It also makes sense that the higher level of abstraction have a different set of options.

`@cypress/webpack-dev-server` and `@cypress/vite-dev-server` should not be changed in this PR.

Reply:
Is there a followup issue to address this? I want to make sure all dev server modules behave consistently. /cc @JessicaSachs